main.rb

```diff
@@ -0,0 +1,98 @@
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    size.times { |i| yield(to_a[i]) }
+    self
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    size.times do |i|
+      yield to_a[i], i
+    end
+    self
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    ary = []
+    my_each do |elem|
+      ary << elem if yield elem
+    end
+    ary
+  end
+
+  def my_all?
+    return nil unless block_given?
+
+    my_each do |elem|
+      return false unless yield elem
+    end
+    true
+  end
+
+  def my_map(proc = nil)
+    return dup unless block_given?
+    return dup if proc.nil?
+
+    ary = []
+    if proc.nil?
+      to_a.my_each { |elem| ary << yield(elem) }
+    else
+      to_a.my_each { |elem| ary << proc.call(elem) }
+    end
+    ary
+  end
+
+  def my_none?
+    return nil unless block_given?
+
+    my_each do |elem|
+      return false if yield elem
+    end
+    true
+  end
+
+  def my_any?
+    return nil unless block_given?
+
+    my_each do |elem|
+      return true if yield elem
+    end
+    false
+  end
```

Comment:
- enumerables #my_any? when no block or argument is given returns true if at least one of the collection is not false or nil
- enumerables #my_any? when no block or argument is given returns false if at least one of the collection is not true
- enumerables #my_any? when a class is passed as an argument returns true if at least one of the collection is a member of such class::Numeric
- enumerables #my_any? when a class is passed as an argument returns true if at least one of the collection is a member of such class::Integer
- enumerables #my_any? when a Regex is passed as an argument returns true if any of the collection matches the Regex
- enumerables #my_any? when a Regex is passed as an argument returns false if none of the collection matches the Regex
- enumerables #my_any? when a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern
- enumerables #my_any? when a pattern other than Regex or a Class is given returns true if any of the collection matches the pattern